agent/lib/kontena/network_adapters/weave.rb

```diff
@@ -253,8 +256,9 @@ def launch(node)
       connect_peers(peer_ips)
       info "using trusted subnets: #{trusted_subnets.join(',')}" if trusted_subnets && !already_started?
       post_start(node)
-
+      debug "weave ready, publishing 'network_adapter:start' event"
       Celluloid::Notifications.publish('network_adapter:start', node) unless already_started?
+      Celluloid::Notifications.publish('network_adapter:restart', node) if @recreated
```

Comment:
This can publish both `network_adapter:start` and `network_adapter:restart` if the agent is upgraded to a newer version, so the `Weave` actor intializes without `@already_started `set, but with an outdated `weave` container running?